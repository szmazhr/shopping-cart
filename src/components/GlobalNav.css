.global-nav{
  background-color: rgb(164, 255, 225);
  box-shadow: 0 0 5px 0 rgba(0,0,0,0.5);
  padding: 0 2rem;
  position: sticky;
  top: 0;
  z-index: 100;
}
.global-nav .container{
  display: flex;
  justify-content: space-between;
  align-items: center;
  max-width: 1440px;
  margin: auto;
}
.global-nav .brandName{
  font-size: 2rem;
  font-weight: bold;
}
.global-nav .navbar{
  display: flex;
}
.global-nav ul{
  display: flex;
  list-style: none;
  padding: 0;
  margin: 0;
}

.global-nav li{
  display: flex;
  align-items: center;
  justify-content: center;
  min-width: 70px;
  position: relative;
}
.global-nav a{
  text-decoration: none;
  color: rgb(27, 27, 27);
}
.global-nav .navbar a{
  padding: 1rem;
  display: flex;
  font-size: 1.5rem;
  transition: background-color 0.3s ease-in-out;
  align-items: center;
  justify-content: center;
  height: 100%;
  width: 100%;
}
.global-nav .navbar a img{
  width: 75%;
}
.global-nav .navbar a.active{
  background-color: rgb(82, 255, 197);
}
.global-nav .navbar a:hover{
  background-color: rgb(130, 255, 213);
}
.global-nav li .cart-count{
  position: absolute;
  top: 10px;
  right: 10px;
  background-color: rgb(27, 27, 27);
  color: white;
  height: 22px;
  width: 22px;
  border-radius: 50%;
  display: flex;
  align-items: center;
  justify-content: center;
  pointer-events: none;
}