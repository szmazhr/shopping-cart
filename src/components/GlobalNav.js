import { Link, NavLink, Outlet } from 'react-router-dom';
import cartIcon from '../assets/cart.svg';
import './GlobalNav.css';
import Footer from './Footer';

function GlobalNav({ itemCount }) {
  return (
  <>
  <header className='global-nav'>
    <div className="container">

    <div className="brandName">
      <Link to="/">sZ Shop</Link>
    </div>
    <nav className='navbar'>
      <ul>
        <li>
          <NavLink to="/">Home</NavLink>
        </li>
        <li>
          <NavLink to="/shop">Shop</NavLink>
        </li>
        <li>
          <NavLink to="/cart"><img src={cartIcon} alt="cart"></img></NavLink>
          {itemCount ? <span className="cart-count">{itemCount}</span> : ''}
        </li>
      </ul>
    </nav>
    </div>
  </header>
    <Outlet />
    <Footer />
  </>
  )
}

export default GlobalNav