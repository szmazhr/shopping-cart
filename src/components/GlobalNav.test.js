import { render, screen } from '@testing-library/react';
import GlobalNav from './GlobalNav';
import { MemoryRouter } from 'react-router-dom';

describe('GlobalNav', () => {
  it('renders correctly', () => {
    render(<GlobalNav/>, { wrapper: MemoryRouter });

    expect(screen.getByText('sZ Shop')).toBeInTheDocument();
    expect(screen.getByRole('list')).toBeInTheDocument();
    expect(screen.getAllByRole('listitem').length).toBe(3);
    expect(screen.getByText('Home')).toBeInTheDocument();
    expect(screen.getByText('Shop')).toBeInTheDocument();
    expect(screen.getByAltText('cart')).toBeInTheDocument();
    expect(screen.queryByText('5')).not.toBeInTheDocument();
  });

  it('renders count correctly', () => {
    render(<GlobalNav itemCount={5}/>, { wrapper: MemoryRouter });
    expect(screen.getByRole('list')).toBeInTheDocument();
    expect(screen.getAllByRole('listitem').length).toBe(3);
    expect(screen.getByText('5')).toBeInTheDocument();
  });

  it('matches snapshot', () =>{
    const {asFragment} = render(<GlobalNav/>, { wrapper: MemoryRouter });
    expect(asFragment()).toMatchSnapshot();
  })
  it('matches snapshot with count', () =>{
    const {asFragment} = render(<GlobalNav itemCount={5} />, { wrapper: MemoryRouter });
    expect(asFragment()).toMatchSnapshot();
  })
});