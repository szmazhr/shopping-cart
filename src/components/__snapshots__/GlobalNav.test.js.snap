// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GlobalNav matches snapshot 1`] = `
<DocumentFragment>
  <header
    class="global-nav"
  >
    <div
      class="container"
    >
      <div
        class="brandName"
      >
        <a
          href="/"
        >
          sZ Shop
        </a>
      </div>
      <nav
        class="navbar"
      >
        <ul>
          <li>
            <a
              aria-current="page"
              class="active"
              href="/"
            >
              Home
            </a>
          </li>
          <li>
            <a
              class=""
              href="/shop"
            >
              Shop
            </a>
          </li>
          <li>
            <a
              class=""
              href="/cart"
            >
              <img
                alt="cart"
                src="cart.svg"
              />
            </a>
          </li>
        </ul>
      </nav>
    </div>
  </header>
  <footer
    class="footer"
  >
    <div>
      <span>
        Made with 
        <img
          alt="heart"
          src="heart.svg"
        />
         by  
        <a
          href="https://github.com/szmazhr"
          rel="noreferrer"
          target="_blank"
        >
          Shahzar Mazhar
        </a>
      </span>
    </div>
    <div
      class="repo_link"
    >
      <a
        href="https://github.com/szmazhr/shopping-cart"
        rel="noreferrer"
        target="_blank"
      >
        <img
          alt="github"
          src="github.svg"
          width="30"
        />
      </a>
    </div>
  </footer>
</DocumentFragment>
`;

exports[`GlobalNav matches snapshot with count 1`] = `
<DocumentFragment>
  <header
    class="global-nav"
  >
    <div
      class="container"
    >
      <div
        class="brandName"
      >
        <a
          href="/"
        >
          sZ Shop
        </a>
      </div>
      <nav
        class="navbar"
      >
        <ul>
          <li>
            <a
              aria-current="page"
              class="active"
              href="/"
            >
              Home
            </a>
          </li>
          <li>
            <a
              class=""
              href="/shop"
            >
              Shop
            </a>
          </li>
          <li>
            <a
              class=""
              href="/cart"
            >
              <img
                alt="cart"
                src="cart.svg"
              />
            </a>
            <span
              class="cart-count"
            >
              5
            </span>
          </li>
        </ul>
      </nav>
    </div>
  </header>
  <footer
    class="footer"
  >
    <div>
      <span>
        Made with 
        <img
          alt="heart"
          src="heart.svg"
        />
         by  
        <a
          href="https://github.com/szmazhr"
          rel="noreferrer"
          target="_blank"
        >
          Shahzar Mazhar
        </a>
      </span>
    </div>
    <div
      class="repo_link"
    >
      <a
        href="https://github.com/szmazhr/shopping-cart"
        rel="noreferrer"
        target="_blank"
      >
        <img
          alt="github"
          src="github.svg"
          width="30"
        />
      </a>
    </div>
  </footer>
</DocumentFragment>
`;
